extensions/standard-processors/processors/SplitContent.cpp

```diff
@@ -0,0 +1,171 @@
+    session.appendBuffer(splits.back(), buffer);
+  }
+}
+
+namespace {
+std::shared_ptr<core::FlowFile> createFirstSplit(core::ProcessSession& session) {
+  auto first_split = session.create();
+  if (!first_split) {
+    throw Exception(PROCESSOR_EXCEPTION, "Couldn't create FlowFile");
+  }
+  return first_split;
+}
+
+void updateSplitAttributesAndTransfer(core::ProcessSession& session, const std::vector<std::shared_ptr<core::FlowFile>>& splits, const core::FlowFile& original) {
+  const std::string fragment_identifier_ = utils::IdGenerator::getIdGenerator()->generate().to_string();
+  for (size_t split_i = 0; split_i < splits.size(); ++split_i) {
+    const auto& split = splits[split_i];
+    split->setAttribute(SplitContent::FragmentCountOutputAttribute.name, std::to_string(splits.size()));
+    split->setAttribute(SplitContent::FragmentIndexOutputAttribute.name, std::to_string(split_i + 1));  // One based indexing
+    split->setAttribute(SplitContent::FragmentIdentifierOutputAttribute.name, fragment_identifier_);
+    split->setAttribute(SplitContent::SegmentOriginalFilenameOutputAttribute.name, original.getAttribute(core::SpecialFlowAttribute::FILENAME).value_or(""));
+    session.transfer(split, SplitContent::Splits);
+  }
+}
+
+bool lastSplitIsEmpty(const std::vector<std::shared_ptr<core::FlowFile>>& splits) {
+  return splits.back()->getSize() == 0;
+}
+}  // namespace
+
+void SplitContent::onTrigger(core::ProcessContext& context, core::ProcessSession& session) {
+  gsl_Assert(!byte_sequence_.empty());
+  const auto original = session.get();
+  if (!original) {
+    context.yield();
+    return;
+  }
+
+  const auto ff_content_stream = session.getFlowFileContentStream(*original);
+  if (!ff_content_stream) {
+    throw Exception(PROCESSOR_EXCEPTION, fmt::format("Couldn't access the ContentStream of {}", original->getUUID().to_string()));
+  }
+  std::vector<std::byte> buffer{};
+  buffer.reserve(BUFFER_TARGET_SIZE + byte_sequence_.size());
+  size_t matching_bytes = 0;
+  bool ended_with_byte_sequence = false;
+  std::vector<std::shared_ptr<core::FlowFile>> splits{};
+  splits.push_back(createFirstSplit(session));
+
+  while (auto latest_byte = ff_content_stream->readByte()) {
+    buffer.push_back(*latest_byte);
+    if (ended_with_byte_sequence) {
+      ended_with_byte_sequence = false;
+      if (!lastSplitIsEmpty(splits)) {
+        splits.push_back(createNewSplit(session));
+      } else if (keep_byte_sequence && byte_sequence_location_ == ByteSequenceLocation::Leading) {
+        session.appendBuffer(splits.back(), byte_sequence_);
+      }
+    }
+    if (latest_byte == byte_sequence_[matching_bytes]) {
+      matching_bytes++;
+      if (matching_bytes == byte_sequence_.size()) {
+        // Found the Byte Sequence
+        finalizeLatestSplitContent(session, splits.back(), buffer);
+        ended_with_byte_sequence = true;
+        matching_bytes = 0;
+        buffer.clear();
+      }
+    } else {
+      matching_bytes = 0;
```

Comment:
this could fail for the following case: byte_sequence_ = "ab" and content = "aab", when the matching fails for the second 'a' it starts from the beginning for the next character

Reply:
thanks, I've implemented your idea (talked about it in private), https://github.com/apache/nifi-minifi-cpp/pull/1872/commits/66435037a5a85eb381936ab3a0c708e44c13a73c